monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,271 @@
+    the larger the grey scale value. The grey value of the instance's border will be larger.
+
+    Args:
+        kernel_size: the size of the Sobel kernel. Defaults to 21.
+        remove_small_objects: whether need to remove some objects in segmentation results. Defaults to True.
+        min_size: objects smaller than this size are removed if `remove_small_objects` is True. Defaults to 10.
+
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        kernel_size: int = 21,
+        min_size: int = 10,
+        remove_small_objects: bool = True,
+    ) -> None:
+        self.sobel_gradient =  SobelGradients(kernel_size=kernel_size)
+        if remove_small_objects:
+            self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        else:
+            self.remove_small_objects = None  # type: ignore
+
+    def __call__(self, mask, hover_map):
+        """
+        Args:
+            mask: binarized segmentation result.  Shape must be [1, H, W].
+            hover_map:  horizontal and vertical distances of nuclear pixels to their centres of mass. Shape must be [2, H, W].
+                The first and second channel represent the horizontal and vertical maps respectively. For more details refer
+                to papers: https://arxiv.org/abs/1812.06499.
+
+        Return:
+            Foreground probability map.
+        """
+        mask = convert_to_tensor(mask, track_meta=get_track_meta())
+        hover_map = convert_to_tensor(hover_map, track_meta=get_track_meta())
+
+        hover_h = hover_map[0:1, ...]
+        hover_v = hover_map[1:2, ...]
+
+        hover_h = (hover_h - np.min(hover_h)) / (np.max(hover_h) - np.min(hover_h))
+        hover_v = (hover_v - np.min(hover_v)) / (np.max(hover_v) - np.min(hover_v))
+        sobelh = self.sobel_gradient(hover_h)[0, ...]
+        sobelv = self.sobel_gradient(hover_v)[1, ...]
+        sobelh = 1 - (sobelh - np.min(sobelh)) / (np.max(sobelh) - np.min(sobelh))
+        sobelv = 1 - (sobelv - np.min(sobelv)) / (np.max(sobelv) - np.min(sobelv))
+
+        # combine the h & v values using max
+        overall = np.maximum(sobelh, sobelv)
+        overall = overall - (1 - mask)
+        overall[overall < 0] = 0
+
+        return convert_to_dst_type(overall, mask, dtype=np.float32)[0]
+
+
+class GenerateDistanceMap(Transform):
+    """
+    Generate distance map.
+    In general, the instance map is calculated from the distance to the background. Here, we use 1 - "foreground probability map" to
+    generate the distance map. Nnuclei values form mountains so inverse to get basins.
+
+    Args:
+        sigma: std. Used in `GaussianFilter`. Could be a single value, or `spatial_dims` number of values. Defaults to 0.4.
+        smooth_fn: execute smooth function on distance map. Defaults to None. You can specify "guassian" or other callable functions.
```

Comment:
Could you please help list all the supported `str` values?

Thanks.

Reply:
Have updated docstring,  now it only supports callable functions and uses `GaussianSmooth` as default. Thanks!